app/components/wallet/WalletTrezorDialog.js

```diff
@@ -0,0 +1,744 @@
+      TrezorConnect.dispose();
+    }
+  }
+
+  /**
+   * prepares and updates the UI state
+   */
+  _updateState = async () => {
+    const { intl } = this.context;
+
+    switch (this.progressState) {
+      case ProgressStateOption.ABOUT:
+        this.state.action_btn_processing = false;
+        this.state.action_btn_name = intl.formatMessage(messages.nextButtonLabel);
+        this.state.currentProgressStepInfo.currentIndex = ProgressStep.ABOUT;
+        this.state.currentProgressStepInfo.error = false;
+        this.state.error_or_live_info_text = '';
+        break;
+      case ProgressStateOption.CONNECT_LOAD:
+        this.state.action_btn_processing = false;
+        this.state.action_btn_name = intl.formatMessage(messages.connectButtonLabel);
+        this.state.currentProgressStepInfo.currentIndex = ProgressStep.CONNECT;
+        this.state.currentProgressStepInfo.error = false;
+        this.state.error_or_live_info_text = '';
+        break;
+      case ProgressStateOption.CONNECT_START:
+        this.state.action_btn_processing = true;
+        this.state.action_btn_name = intl.formatMessage(messages.connectButtonLabel);
+        this.state.currentProgressStepInfo.currentIndex = ProgressStep.CONNECT;
+        this.state.currentProgressStepInfo.error = false;
+        this.state.error_or_live_info_text = intl.formatMessage(messages.connectMsgCheckingTrezor);
+        break;
+      case ProgressStateOption.CONNECT_ERROR:
+        this.state.action_btn_processing = false;
+        this.state.action_btn_name = intl.formatMessage(messages.connectButtonLabel);
+        this.state.currentProgressStepInfo.currentIndex = ProgressStep.CONNECT;
+        this.state.currentProgressStepInfo.error = true;
+        // eslint-disable-next-line max-len
+        this.state.error_or_live_info_text = intl.formatMessage(this.trezorDeviceInfo.error);
+        break;
+      case ProgressStateOption.SAVE_LOAD:
+        this.state.action_btn_processing = false;
+        this.state.action_btn_name = intl.formatMessage(messages.saveButtonLabel);
+        this.state.currentProgressStepInfo.currentIndex = ProgressStep.SAVE;
+        this.state.currentProgressStepInfo.error = false;
+        this.form.$('walletName').value = this.trezorDeviceInfo.features.label;
+        this.state.error_or_live_info_text = '';
+        break;
+      case ProgressStateOption.SAVE_START:
+        this.state.action_btn_processing = true;
+        this.state.action_btn_name = intl.formatMessage(messages.saveButtonLabel);
+        this.state.currentProgressStepInfo.currentIndex = ProgressStep.SAVE;
+        this.state.currentProgressStepInfo.error = false;
+        this.state.error_or_live_info_text = '';
+        break;
+      case ProgressStateOption.SAVE_ERROR:
+        // managed in render()
+        break;
+      default:
+        console.error(`[TREZOR] Error state, handle _updateState for: ${this.progressState}`);
+        break;
+    }
+    this.setState({});
```

Comment:
as we discussed we should not do `setState({})` but rather an object created from the switch

Reply:
https://github.com/Emurgo/yoroi-frontend/commit/41cfb4eb871cfd6b52c9907b2799bb2155821bfb

Reply:
the change regarding storing walletName in state is not pushed yet, will push it very soon.

Reply:
https://github.com/Emurgo/yoroi-frontend/commit/55ee34aab9fde294b6024489f17b3436301ca331

Reply:
https://github.com/Emurgo/yoroi-frontend/commit/f51d20b5cc73f3304169a159cb891c22122451b0